### PR TITLE
Add symbol plugin

### PR DESCRIPTION
**Pull request by tobrun**
This PR adds integration of https://github.com/mapbox/mapbox-gl-native/tree/tvn-code-gen-symbols as separate plugion so we can kick the tires on it. Initially this annotation plugin only contains code for showing a symbol, next steps would be to include fill/circle/line support.

### CODE REVIEW

**Comment by tobrun** [comment]
Tests were added as part of https://github.com/mapbox/mapbox-plugins-android/pull/642/commits/b86b4713c8561fae8300a97e2e7f381861ad340d

File: app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/annotation/SymbolManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+  @Test
+  public void testSymbolSpacingAsConstant() {
+    validateTestSetup();
+    setupSymbolManager();
+    Timber.i("symbol-spacing");
+    invoke(mapboxMap, (uiController, mapboxMap) -> {
+      assertNotNull(symbolManager);
+
+      symbolManager.setSymbolSpacing(0.3f);
+      assertEquals((Float) symbolManager.getSymbolSpacing(), (Float) 0.3f);
```
Comment (LukasPaczos) [inline app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/annotation/SymbolManagerTest.java]: This and other casts are redundant.

File: app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/annotation/SymbolManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+  @Test
+  public void testSymbolAvoidEdgesAsConstant() {
+    validateTestSetup();
+    setupSymbolManager();
+    Timber.i("symbol-avoid-edges");
+    invoke(mapboxMap, (uiController, mapboxMap) -> {
+      assertNotNull(symbolManager);
+
+      symbolManager.setSymbolAvoidEdges(true);
+      assertEquals((Boolean) symbolManager.getSymbolAvoidEdges(), (Boolean) true);
```
[the remark this answers is not in the document]
Comment (tobrun) [reply app/src/androidTest/java/com/mapbox/mapboxsdk/plugins/annotation/SymbolManagerTest.java]: this is generated code, same outcome

File: plugin-annotation/src/main/java/com/mapbox/mapboxsdk/plugins/annotation/SymbolManager.java [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+  public List<Symbol> getSymbols() {
+    return symbolList;
+  }
+
+  /**
+   * Trigger an update to the underlying source
+   */
+  public void updateSource() {
+    // todo move feature creation to a background thread?
+    List<Feature> features = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (tobrun) [reply plugin-annotation/src/main/java/com/mapbox/mapboxsdk/plugins/annotation/SymbolManager.java]: sure

**Pull request merged**